lib/jnpr/junos/utils/sw.py

```diff
@@ -979,29 +985,40 @@ def reboot(self, in_min=0, at=None, all_re=True, on_node=None):
             else:
                 cmd = E('request-node-reboot')
                 cmd.append(E('node', on_node))
+        elif vmhost is True:
+            cmd = E('request-vmhost-reboot')
         else:
             cmd = E('request-reboot')
-            if all_re is True:
-                if self._multi_RE is True and self._multi_VC is False:
-                    cmd.append(E('both-routing-engines'))
-                elif self._mixed_VC is True:
-                    cmd.append(E('all-members'))
+        if all_re is True:
+            if vmhost is True:
+                cmd.append(E('routing-engine', 'both'))
+            elif self._multi_RE is True and self._multi_VC is False:
+                cmd.append(E('both-routing-engines'))
+            elif self._mixed_VC is True:
+                cmd.append(E('all-members'))
         if in_min >= 0 and at is None:
             cmd.append(E('in', str(in_min)))
         else:
             cmd.append(E('at', str(at)))
         try:
-            rsp = self.rpc(cmd)
+            rsp = self.rpc(cmd, ignore_warning=True, normalize=True)
             if self._dev.facts['_is_linux']:
                 got = rsp.text
             else:
                 got = rsp.getparent().findtext('.//request-reboot-status').strip()
+                if got is None:
+                    # Oon some platforms stopping/rebooting both
+                    # REs produces <output> messages and
+                    # <request-reboot-status> messages.
+                    output_msg = '\n'.join([i.text for i in got.findall('output')
+                                            if i.text is not None])
+                    if output_msg is not '':
+                        got = output_msg
             return got
```

Comment:
Will `got` be equal to `None` in case Junos device does not output anything ??

Can we change it, or should we continue with `None` to be backward-compatible ??

Reply:
I have removed split as can be an issue when findtext returns None. https://github.com/Juniper/py-junos-eznc/pull/952/commits/fc6cf16a7c66a5ac0d950b920e59bf698a51295e#diff-d1aa66cff516a1461b58eda901c5a52dR1007

got should have some text as we are handling None case where we check for  `<output>`